src/sentry/debug_files/artifact_bundle_indexing.py

```diff
@@ -167,9 +174,24 @@ def get_flat_file_meta(self) -> Optional[FlatFileMeta]:
                     "artifact_bundle_flat_file_indexing.flat_file_meta.db_miss",
                     tags={"meta_type": meta_type},
                 )
-                return None
+                meta = FlatFileMeta.build_none()
+            else:
+                metrics.incr(
+                    "artifact_bundle_flat_file_indexing.flat_file_meta.db_hit",
+                    tags={"meta_type": meta_type},
+                )
 
+            # We want to cache in both cases, either a value is found or a value was not found.
             self.set_flat_file_meta_in_cache(meta)
+        else:
+            metrics.incr(
+                "artifact_bundle_flat_file_indexing.flat_file_meta.cache_hit",
+                tags={"meta_type": meta_type},
+            )
+
+        # In case the meta that we found was none, we want to return None.
+        if meta.is_none():
```

Comment:
At first I though, why do we have a sentinal value for `FlatFileMeta` instead of just dealing with `None` directly in the get/set from cache. But yeah, you need a sentinel value to distinguish "not in cache" vs "None in cache". :-D

Reply:
Exactly, this is why I did it.